src/lib/routing/html-id-slugger.ts

```diff
@@ -0,0 +1,18 @@
+import GithubSlugger from 'github-slugger';
```

Comment:
This file should be either a lib here or in the provider component, but not both :)